reya-rust-sdk/src/http_provider.rs

```diff
@@ -213,6 +207,89 @@ impl HttpProvider {
         eyre::Ok(receipt.transaction_hash)
     }
 
+    ///
+    /// execute a batch of orders
+    ///
+    pub async fn execute_batch(
+        &self,
+        signer: PrivateKeySigner,
+        batch_orders: &mut Vec<data_types::BatchOrder>,
+    ) -> eyre::Result<B256> // return the transaction hash
+    {
+        //
+        let mut orders: Vec<CoreProxy::ConditionalOrderDetails> = vec![];
+        let mut signatures: Vec<CoreProxy::EIP712Signature> = vec![];
+
+        // create http provider
+        let provider = ProviderBuilder::new()
+            .with_recommended_fillers()
+            .wallet(EthereumWallet::from(signer))
+            .on_http(self.sdk_config.rpc_sdk_url.clone());
+
+        let core_proxy = CoreProxy::new(
+            self.sdk_config.order_gateway_contract_address.parse()?,
+            provider.clone(),
+        );
+
+        // add all order ans signatures to the batch vector
+        for batch_order in batch_orders {
+            //trace!("Batch execute orders {}", batch_order);
+
+            let mut encoded_inputs: Vec<u8> = Vec::new();
+            if batch_order.order_type == data_types::OrderType::StopLoss {
+                // generate encoded core command for the input bytes of a stop_loss order
+                // The input byte structure is:
+                // {
+                //     trigger_price, // stop_price!
+                //     price_limit,   // price limit is the slippage tolerance,we can set it to max uint or zero for now depending on the direction of the trade
+                // }// endcoded
+                let trigger_price = batch_order.stop_price;
+                let bytes = (trigger_price, batch_order.price_limit).abi_encode_sequence();
+                encoded_inputs.clone_from(&bytes);
+            }
+
+            let counterparty_account_ids: Vec<u128> = vec![2u128]; // hardcode counter party id = 2
+
+            orders.push(CoreProxy::ConditionalOrderDetails {
+                accountId: batch_order.account_id,
+                marketId: batch_order.market_id,
+                exchangeId: batch_order.exchange_id,
+                counterpartyAccountIds: counterparty_account_ids.clone(),
+                orderType: batch_order.order_type as u8,
+                inputs: Bytes::from(encoded_inputs),
+                signer: batch_order.signer_address,
+                nonce: batch_order.order_nonce,
+            });
+
+            signatures.push(batch_order.eip712_signature.clone());
+        }
+
+        let builder = core_proxy.batchExecute(orders, signatures);
+        let transaction_result = builder.send().await?;
+
+        let receipt = transaction_result.get_receipt().await?;
```

Comment:
btw, one thing we'll introduce with @0xZenus is we will additionally also emit events on-chain for failed executions within the batch and they should be available as logs in the receipt